src/custom_nodes/add_one/add_one.cpp

```diff
@@ -46,74 +48,31 @@ int initializeInternalManager(void** customNodeLibraryInternalManager, const str
-    InternalManager* internalManager = static_cast<InternalManager*>(*customNodeLibraryInternalManager);
-    NODE_ASSERT(internalManager != nullptr, "internalManager is not initialized");
-    std::unique_lock<std::shared_timed_mutex> lock(internalManager->getInternalManagerLock());
-
-    int output_queue_size = get_int_parameter("output_queue_size", params, paramsCount, internalManager->getCurrentOutputQueueSize());
-    NODE_ASSERT(output_queue_size > 0, "output_queue_size should be greater than 0");
-
-    int info_queue_size = get_int_parameter("info_queue_size", params, paramsCount, internalManager->getCurrentInfoQueueSize());
-    NODE_ASSERT(info_queue_size > 0, "info_queue_size should be greater than 0");
-
-    if (internalManager->getCurrentOutputQueueSize() != output_queue_size) {
-        NODE_ASSERT(internalManager->recreateBuffersQueue(OUTPUT_NAME, 1 * sizeof(CustomNodeTensor), output_queue_size), "output buffer recreation failed");
-
-        uint64_t byteSize = sizeof(float) * internalManager->getOutputSize();
-        NODE_ASSERT(internalManager->recreateBuffersQueue(OUTPUT_TENSOR_NAME, byteSize, output_queue_size), "output tensor buffer recreation failed");
-        NODE_ASSERT(internalManager->recreateBuffersQueue(OUTPUT_TENSOR_DIMS_NAME, 2 * sizeof(uint64_t), output_queue_size), "output tensor dims buffer recreation failed");
-
-        internalManager->setCurrentOutputQueueSize(output_queue_size);
-    }
-
-    if (internalManager->getCurrentInfoQueueSize() != info_queue_size) {
-        NODE_ASSERT(internalManager->recreateBuffersQueue(INPUT_INFO_NAME, 1 * sizeof(CustomNodeTensorInfo), info_queue_size), "input info buffer recreation failed");
-        NODE_ASSERT(internalManager->recreateBuffersQueue(OUTPUT_INFO_NAME, 1 * sizeof(CustomNodeTensorInfo), info_queue_size), "output info buffer recreation failed");
-
-        NODE_ASSERT(internalManager->recreateBuffersQueue(INPUT_INFO_DIMS_NAME, 2 * sizeof(uint64_t), info_queue_size), "input info dims buffer recreation failed");
-
-        NODE_ASSERT(internalManager->recreateBuffersQueue(OUTPUT_INFO_DIMS_NAME, 2 * sizeof(uint64_t), info_queue_size), "output info dims buffer recreation failed");
-
-        internalManager->setCurrentInfoQueueSize(info_queue_size);
-    }
-
-    return 0;
-}
-
-int initialize(void** customNodeLibraryInternalManager, const struct CustomNodeParam* params, int paramsCount) {
-    auto status = 0;
-    if (*customNodeLibraryInternalManager == nullptr) {
-        status = initializeInternalManager(customNodeLibraryInternalManager, params, paramsCount);
-    } else {
-        status = reinitializeInternalManagerIfNeccessary(customNodeLibraryInternalManager, params, paramsCount);
-    }
-    NODE_ASSERT(status == 0, "initialize failed");
-    return 0;
-}
-
 int deinitialize(void* customNodeLibraryInternalManager) {
-    if (customNodeLibraryInternalManager != nullptr) {
```

Comment:
Just a question, what has changed that made this if uneccessary?

Reply:
There is no particular change, however I came into conclusion that if in initialize the library creates customNodeLibraryInternalManager than there should be no case that it is nullptr.